homeassistant/components/ecobee/climate.py

```diff
@@ -176,18 +176,31 @@ async def async_setup_entry(hass, config_entry, async_add_entities):
     """Set up the ecobee thermostat."""
 
     data = hass.data[DOMAIN]
+    entities = []
 
-    devices = [Thermostat(data, index) for index in range(len(data.ecobee.thermostats))]
+    for index in range(len(data.ecobee.thermostats)):
+        thermostat = data.ecobee.get_thermostat(index)
+        if not thermostat["modelNumber"] in ECOBEE_MODEL_TO_NAME:
+            _LOGGER.error(
+                "Model number for ecobee thermostat %s not recognized. "
+                "Please visit this link to open a new issue: "
+                "https://github.com/home-assistant/core/issues "
+                "and include the following information: "
+                "Unrecognized model number: %s",
+                thermostat["name"],
+                thermostat["modelNumber"],
+            )
+        entities.append(Thermostat(data, index))
```

Comment:
If we pass the thermostat to the entity we save a call in the entity init.
```suggestion
        entities.append(Thermostat(data, index, thermostat))
```